include/engine/datafacade/datafacade_base.hpp

```diff
@@ -50,6 +50,14 @@ class BaseDataFacade
 {
   public:
     using RTreeLeaf = extractor::EdgeBasedNodeSegment;
+
+    template <typename T>
+    using RangeT = boost::any_range<T, boost::random_access_traversal_tag, const T, std::ptrdiff_t>;
+    using NodesIDRangeT = RangeT<NodeID>;
+    using WeightsRangeT = RangeT<SegmentWeight>;
+    using DurationsRangeT = RangeT<SegmentDuration>;
+    using DatasourceIDRangeT = RangeT<DatasourceID>;
```

Comment:
Using the type extension `T` kind of breaks convention were we would use that for template parameter names only. We could use `Range`, `NodesIDRange` etc instead.